client/my-sites/plugins/marketplace/controller.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import { getSiteFragment } from 'calypso/lib/route';
+import AsyncLoad from 'calypso/components/async-load';
+import MarketplacePluginDetails from 'calypso/my-sites/plugins/marketplace/marketplace-plugin-details';
+
+export function renderMarketplacePlugin( context, next ) {
+	const siteUrl = getSiteFragment( context.path );
+
+	const pluginSlug = decodeURIComponent( context.params.plugin );
+
+	context.primary = (
+		<MarketplacePluginDetails
+			path={ context.path }
+			marketplacePluginSlug={ pluginSlug }
+			siteUrl={ siteUrl }
+		/>
+	);
+
+	next();
+}
+
+export function renderDomainsPage( context, next ) {
+	context.primary = (
+		<AsyncLoad require="calypso/my-sites/plugins/marketplace/marketplace-domain-upsell" />
+	);
+	next();
+}
+
+export function renderPluginsSetupStatusPage( context, next ) {
+	context.primary = (
+		<AsyncLoad require="calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status" />
+	);
+	next();
+}
+
+export function renderMarketplaceThankYou( context, next ) {
+	// context.primary = (
+	// 	<AsyncLoad require="calypso/my-sites/plugins/marketplace/marketplace-stand-alone-thank-you" />
```

Comment:
Feel free to get rid of any commented code here, I believe it brings little to no value.

Reply:
This change was done inside an already large PR #53107 as I mentioned above. I cherry picked this change because introducing a section is kind of a big deal.So this code is to be rebased with that PR and the code commented out is required.

Reply:
Sounds good as long as it's not still commented in 2022 🤣

Reply:
🤣 Hehe hopefully 🤣 
Actually you have my word 💯